packages/loader/container-loader/src/container.ts

```diff
@@ -880,20 +880,20 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
             // post summary here
             if (hasAttachmentBlobs) {
-                const appSummary: ISummaryTree = this.context.createSummary();
+                const appSummary: ISummaryTree = this.context.createSummary(new Map<string, string>());
                 const protocolSummary = this.captureProtocolSummary();
                 summary = combineAppAndProtocolSummary(appSummary, protocolSummary);
 
                 this._attachState = AttachState.Attaching;
                 this.context.notifyAttaching();
 
-                await this.storageService.uploadSummaryWithContext(summary, {
+                await (this.storageService as any).uploadSummaryWithContext(summary, {
```

Comment:
can you please add "// back-compat" comment with some explanation and version number when cast is expected to be removed, such that we can find it and remove it in the future?